akka-http-core/src/test/scala/akka/http/impl/engine/ws/WebSocketServerReceiveIdleTimeoutSpec.scala

```diff
@@ -0,0 +1,57 @@
+package akka.http.impl.engine.ws
+
+import akka.Done
+import akka.http.impl.util.AkkaSpecWithMaterializer
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model.AttributeKeys.webSocketUpgrade
+import akka.http.scaladsl.model.Uri.apply
+import akka.http.scaladsl.model.ws._
+import akka.stream.scaladsl._
+import akka.stream.testkit._
+import akka.testkit._
+import org.scalatest.concurrent.Eventually
+
+import scala.concurrent.duration.DurationInt
+import scala.concurrent.{Await, Promise}
+import scala.util.{Failure, Success}
+
+class WebSocketServerReceiveIdleTimeoutSpec extends AkkaSpecWithMaterializer(
+  """
+     akka {
+       stream.materializer.debug.fuzzing-mode=off
+       http.server.websocket.log-frames = on
+       http.client.websocket.log-frames = on
+       http.server.websocket.receive-idle-timeout = 1s
+     }
+  """) with Eventually {
+
+  "A WebSocket server" must {
+
+    "terminate the handler flow with an error when elements are not received within receive-idle-timeout" in Utils.assertAllStagesStopped {
+      import system.dispatcher
+      val handlerTermination = Promise[Done]()
+      val handler = Flow[Message].map(identity).watchTermination() { (_, terminationFuture) =>
+        terminationFuture.onComplete {
+          case Success(_) => handlerTermination.trySuccess(Done)
+          case Failure(exception) => handlerTermination.tryFailure(exception)
+        }
+      }
+
+      val bindingFuture = Http().newServerAt("localhost", 0)
+        .bindSync({
+          _.attribute(webSocketUpgrade).get.handleMessages(handler.recover { case ex => {
+            handlerTermination.failure(ex)
+            TextMessage("dummy")
+          }
+          }, None)
+        })
+      val binding = Await.result(bindingFuture, 3.seconds.dilated)
+      val myPort = binding.localAddress.getPort
+
+      Source.maybe.via(Http().webSocketClientFlow(WebSocketRequest("ws://127.0.01:" + myPort))).to(Sink.ignore).run()
+
+      handlerTermination.future.failed.futureValue
```

Comment:
What is the type of exception you get?

Reply:
`akka.stream.StreamIdleTimeoutException`
I've added exception class matchers in tests